src/OpenTelemetry/Trace/Internal/BlankSpan.cs

```diff
@@ -69,9 +67,8 @@ public override ISpanId ParentSpanId
 
         public override bool HasEnded => true;
 
-        public override void SetAttributes(IDictionary<string, IAttributeValue> attributes)
-        {
-        }
+        public override bool IsRecordingEvents => throw new NotImplementedException();
```

Comment:
why throw? I think we should just return false.

typical instrumentation in the lib will call this method and it will throw in the absence of OpenTelemetry implementation.

I think there should be a rule that nothing in the Noop implementaitons throw ever.

Reply:
oversight. You are totally correct